### DISCUSSION TITLE
Mass-creation of tree-based models with MPTT (Region, TenantGroup, InventoryItem, RackGroup, ...?)

### DISCUSSION THREAD

**Discussion by Kircheneer** [nautobot/nautobot#3433] [Show and tell]
Despite MPTT being migrated away from in 2.0 I wanted to briefly shared a workaround I had to use for performance issues when mass-creating regions (the same probably applies to the other MPTT-based models, although I didn't test it):

```
with Region.objects.disable_mptt_updates:
    create_regions()
Region.objects.rebuild()
```

This significantly speeds up region creation (from ~0.5s per region to ~0.02s per region) when creating a lot of them at once by disabling the tree updates and doing them in bulk when all region objects (whose tree relationships are broken prior to the rebuilding) are in place.

**Comment by jathanism** [comment]
Thank you for sharing this @Kircheneer!
